aiida/cmdline/commands/node.py

```diff
@@ -232,79 +231,69 @@ def cat(self, *args):
             else:
                 raise
 
-
-class _Show(VerdiCommand):
-    """
-    Show node information (pk, uuid, class, inputs and outputs)
-    """
+class _Show(VerdiCommandWithSubcommands):
 
     def run(self, *args):
-        """
-        Show node information.
-        """
-        import argparse
-        from aiida.common.exceptions import NotExistent
-
-        parser = argparse.ArgumentParser(
-            prog=self.get_full_command_name(),
-            description='Show information of a node.')
-        parser.add_argument('pk', type=int, default=None, nargs="+",
-                            help="ID of the node.")
-        parser.add_argument('--print-groups', action='store_true',
-                            dest='print_groups', default=False,
-                            help="Show groups containing the nodes.")
-        parser.add_argument('--no-print-groups', '--dont-print-groups',
-                            action='store_false', dest='print_groups',
-                            default=False,
-                            help="Do not show groups containing the nodes"
-                                 "output. Default behaviour.")
-
-        args = list(args)
-        parsed_args = parser.parse_args(args)
-
-        if not is_dbenv_loaded():
-            load_dbenv()
-
-        for pk in parsed_args.pk:
-            try:
-                n = load_node(pk)
-                self.print_node_info(n, print_groups=parsed_args.print_groups)
-            except NotExistent as e:
-                print >> sys.stderr, e.message
-                sys.exit(1)
+        verdi()
+
+@node.command('show')
+@click.argument('identifiers', type=str, nargs=-1)
+@click.option('--uuid', 'uuid', flag_value=True,
+              help="If activated the identifier is UUID, "
+                   "otherwise it is a pk/id..")
+@click.option('--print-groups', 'print_groups', flag_value=True,
+              help="Show groups containing the nodes.")
+@click.option('--no-print-groups', '--dont-print-groups', 'print_groups',
+              flag_value=False, help="Do not show groups containing the nodes "
+                                     "output. Default behaviour.")
```

Comment:
Just my curiosity, how do we know which of the two defines the default behaviour?

Reply:
Shouldn't this just be a single option? Either you want to print the groups or you don't? In that case it should just be `@click.option('--print-groups', is_flag=True, help="")`  or if we want the default to be to print the groups, then it would make sense to add an option `@click.option('--no-print-groups', is_flag=True)`. As a side note, if you define long name option, i.e. `--print-groups`, the metavariable will automatically be determined as `print_groups`, so you do not have to explicitly define that.

Reply:
@giovannipizzi 
Just from testing (I didn't have the time to read extensively the doc): if an option is not activated, then it is False. So in a few words --no-print-groups & --dont-print-groups are there for "beautification" reasons and I added them in order to appear at the help menu.

@sphuber 
I agree. I just left it as it was before because I didn't want to open now the discussion if the already existing arguments are reasonable, if they are used in a script etc. I also prefer to explicitly define the print_groups because I prefer to be more clear even if it is too verbose sometimes.

To conclude. It's fine for me choices (to leave '--no-print-groups', '--dont-print-groups' or to remove it)